source/extensions/filters/network/redis_proxy/config.cc

```diff
@@ -57,15 +57,19 @@ Network::FilterFactoryCb RedisProxyFilterConfigFactory::createFilterFactoryFromP
   }
   addUniqueClusters(unique_clusters, prefix_routes.catch_all_route());
 
+  auto redis_command_stats = Common::Redis::RedisCommandStats::createRedisCommandStats(
```

Comment:
Why are we creating the RedisCommandStats objects both here and in redis_cluster.cc:173 ? I think they should both be able to share the same RedisCommandStats objects to avoid creating extra copies of the StatNames.

Can we contrive to share the same object in all production call-sites?

This is not necessary for functional correctness but it would save memory to have them shared.

There's a lot of data structures in play here but one plausible place to hold it is NetworkFilters::Common::Redis::Client::ClientFactory, although poking around there it seems there's a style-guide violating static instance of that. I don't want to boil the ocean here but maybe it'd be OK to lazy-create the RedisCommandStats there so it can be accessed everywhere?

E.g. 
```
RedisCommandStats&
Extensions::NetworkFilters::Common::Redis::Client::ClientFactoryImpl::getOrCreateRedisCommandStats(Stats::SymbolTable& symbol_table) {
  Thread::LockGuard lock(mutex_);
  if (redis_command_stats_ == nullptr) {
    redis_command_stats_ = std::make_unique<RedisCommandStats>(symbol_table);
  }
  return *redis_command_stats_;
}
```

Here I've deviated from using the shared_ptr as with this strict ownership model you don't really need it.

WDYT?

Reply:
I'm also OK with a TODO to resolve this in a follow-up.